deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java

```diff
@@ -200,6 +195,30 @@
     @JsonProperty("stageType")
     private EnvType stage_type;
 
+    public void validate() throws Exception {
+        // A bunch of these fields will always be alphanumeric (with _ and -)
+        String envRegEx = "^[A-Za-z0-9_\\-]*$";
+        if (!this.env_name.matches(envRegEx)) {
+            throw new IllegalArgumentException(String.format("Environment name must match regex %s", envRegEx));
+        }
+        String stageRegEx = "^[A-Za-z0-9_\\-]*$";
+        if (!this.stage_name.matches(stageRegEx)) {
+            throw new IllegalArgumentException(String.format("Stage name must match regex %s", stageRegEx));
+        }
+        String buildRegEx = "^[A-Za-z0-9_\\.\\/\\-]*$";
+        if (!this.build_name.matches(buildRegEx)) {
+            throw new IllegalArgumentException(String.format("Build name must match regex %s", buildRegEx));
+        }
+        String branchRegEx = "^[A-Za-z0-9_\\:\\.\\,\\/\\-]*$";
```

Comment:
nit: could make these static / consts.

could also probably unify / reuse the ones that are the same

Reply:
strings are immutable in java